src/Microsoft.ML.TimeSeries/DTRootCauseLocalizationType.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// Allows a member to be marked as a <see cref="RootCauseLocalizationInputDataViewType"/>, primarily allowing one to set
+    /// root cause localization input.
+    /// </summary>
+    public sealed class RootCauseLocalizationInputTypeAttribute : DataViewTypeAttribute
+    {
```

Comment:
Can you please explain the need for this data type? 

I saw the test code was annotated with this attribute. Does this mean the input data has to be annotated ahead of time for all inputs that need to be considered for root cause localization?

Reply:
I think my answer is Yes, Here is my explanation.

Input and output for root cause are customized class(not only double, string such basic type), according to Transformer design, for customized dataview type, a decorated data view type attribute need to be registered. I follow the example of ImageType and when the decorated class is used , the mapping will be registered. [Here ](https://github.com/dotnet/machinelearning/blob/02a857a7646188fec2d1cba5e187a6c9d0838e23/src/Microsoft.ML.Data/DataView/InternalSchemaDefinition.cs#L194-L195) is the code where ML.net checks the type.
` else if (!itemType.TryGetDataKind(out _) && !DataViewTypeManager.Knows(itemType, attributes))
                throw Contracts.ExceptParam(nameof(rawType), "Could not determine an IDataView type for member {0}", name);`


I have thought of registering the mapping in transformer, but it seems a very hacky way and I haven't found any precedent. If you have better suggestion about the resgitration, feel free to let me know and I will update it.

Reply:
> Can you please explain the need for this data type?
> 
> I saw the test code was annotated with this attribute. Does this mean the input data has to be annotated ahead of time for all inputs that need to be considered for root cause localization?

I think my answer is Yes, Here is my explanation.

Input and output for root cause are customized class(not only double, string such basic type), according to Transformer design, for customized dataview type, a decorated data view type attribute need to be registered. I follow the example of ImageType and when the decorated class is used , the mapping will be registered. [Here ](https://github.com/dotnet/machinelearning/blob/02a857a7646188fec2d1cba5e187a6c9d0838e23/src/Microsoft.ML.Data/DataView/InternalSchemaDefinition.cs#L194-L195) is the code where ML.net checks the type.
` else if (!itemType.TryGetDataKind(out _) && !DataViewTypeManager.Knows(itemType, attributes))
                throw Contracts.ExceptParam(nameof(rawType), "Could not determine an IDataView type for member {0}", name);`


I have thought of registering the mapping in transformer, but it seems a very hacky way and I haven't found any precedent. If you have better suggestion about the resgitration, feel free to let me know and I will update it.